CodeGen/Generators/QuantityJsonFilesParser.cs

```diff
@@ -124,5 +126,246 @@ private static Localization[] GetLocalizationForPrefixUnit(IEnumerable<Localizat
+        private static readonly Dictionary<(string, Prefix), string> BaseUnitPrefixConversions = GetSIPrefixes()
+            .SelectMany(pair => pair.Value.Select(prefix => (pair.Key, prefix))).ToDictionary(tuple => tuple, tuple => tuple.prefix + tuple.Key.ToCamelCase());
+
+        private static IEnumerable<KeyValuePair<string, Prefix[]>> GetSIPrefixes()
+        {
+            return GetAmountOfSubstancePrefixes()
+                .Concat(GetElectricCurrentPrefixes())
+                .Concat(GetMassPrefixes())
+                .Concat(GetLengthPrefixes())
+                .Concat(GetDurationPrefixes());
+        }
+
+        private static IEnumerable<KeyValuePair<string, Prefix[]>> GetAmountOfSubstancePrefixes()
+        {
+            yield return new KeyValuePair<string, Prefix[]>("Mole",
+                [Prefix.Femto, Prefix.Pico, Prefix.Nano, Prefix.Micro, Prefix.Milli, Prefix.Centi, Prefix.Deci, Prefix.Kilo, Prefix.Mega]);
+            yield return new KeyValuePair<string, Prefix[]>("PoundMole", [Prefix.Nano, Prefix.Micro, Prefix.Milli, Prefix.Centi, Prefix.Deci, Prefix.Kilo]);
+        }
+
+        private static IEnumerable<KeyValuePair<string, Prefix[]>> GetElectricCurrentPrefixes()
+        {
+            yield return new KeyValuePair<string, Prefix[]>("Ampere",
+                [Prefix.Femto, Prefix.Pico, Prefix.Nano, Prefix.Micro, Prefix.Milli, Prefix.Centi, Prefix.Kilo, Prefix.Mega]);
+        }
+
+        private static IEnumerable<KeyValuePair<string, Prefix[]>> GetMassPrefixes()
+        {
+            yield return new KeyValuePair<string, Prefix[]>("Gram",
+                [Prefix.Femto, Prefix.Pico, Prefix.Nano, Prefix.Micro, Prefix.Milli, Prefix.Centi, Prefix.Deci, Prefix.Deca, Prefix.Hecto, Prefix.Kilo]);
+            yield return new KeyValuePair<string, Prefix[]>("Tonne", [Prefix.Kilo, Prefix.Mega]);
+            yield return new KeyValuePair<string, Prefix[]>("Pound", [Prefix.Kilo, Prefix.Mega]);
+        }
+
+        private static IEnumerable<KeyValuePair<string, Prefix[]>> GetLengthPrefixes()
+        {
+            yield return new KeyValuePair<string, Prefix[]>("Meter",
+            [
+                Prefix.Femto, Prefix.Pico, Prefix.Nano, Prefix.Micro, Prefix.Milli, Prefix.Centi, Prefix.Deci, Prefix.Deca, Prefix.Hecto, Prefix.Kilo,
+                Prefix.Mega, Prefix.Giga
+            ]);
+            yield return new KeyValuePair<string, Prefix[]>("Yard", [Prefix.Kilo]);
+            yield return new KeyValuePair<string, Prefix[]>("Foot", [Prefix.Kilo]);
+            yield return new KeyValuePair<string, Prefix[]>("Parsec", [Prefix.Kilo, Prefix.Mega]);
+            yield return new KeyValuePair<string, Prefix[]>("LightYear", [Prefix.Kilo, Prefix.Mega]);
+        }
+
+        private static IEnumerable<KeyValuePair<string, Prefix[]>> GetDurationPrefixes()
+        {
+            yield return new KeyValuePair<string, Prefix[]>("Second", [Prefix.Nano, Prefix.Micro, Prefix.Milli]);
+        }
     }
```

Comment:
This is the most iffy part- I had to hard-code the prefixes that are applicable in the base quantities.. Those aren't likely to change, but if we want to make this _properly_ we either need to do pre-parse the 'Length.json', 'Mass.json' etc in order to get their prefixes before the start of the generation (or think of other ways of storing them).